core/src/main/java/org/apache/accumulo/core/file/blockfile/impl/CachableBlockFile.java

```diff
@@ -379,18 +381,35 @@ private BlockRead getBlock(String _lookup, BlockCache cache, BlockLoader loader)
           return new CachedBlockRead(cb, cb.getBuffer());
         }
 
+        loadLock = cache.getLoadLock(_lookup);
       }
-      /**
-       * grab the currBlock at this point the block is still in the data stream
-       *
-       */
-      _currBlock = loader.get();
 
-      /**
-       * If the block is bigger than the cache just return the stream
-       */
-      return cacheBlock(_lookup, cache, _currBlock, loader.getInfo());
+      try {
+        if (loadLock != null) {
+          loadLock.lock();
+
+          // check cache again after getting lock
+          CacheEntry cb = cache.getBlockNoStats(_lookup);
+
+          if (cb != null) {
+            return new CachedBlockRead(cb, cb.getBuffer());
+          }
+        }
 
+        /**
+         * grab the currBlock at this point the block is still in the data stream
+         *
+         */
+        _currBlock = loader.get();
```

Comment:
what is the point of the `_` in `_currBlock`? is there some special meaning?

Reply:
I don't know why its like that.